processes/processes.go

```diff
@@ -117,53 +192,70 @@ func New() *procPlugin {
 }
 
 // Meta returns plugin meta data
-func Meta() *plugin.PluginMeta {
-	return plugin.NewPluginMeta(
-		pluginName,
-		version,
-		plugin.CollectorPluginType,
-		[]string{},
-		[]string{plugin.SnapGOBContentType},
+func Meta() []plugin.MetaOpt {
+	return []plugin.MetaOpt{
 		plugin.ConcurrencyCount(1),
-	)
+	}
 }
 
 // GetMetricTypes returns list of available metrics
-func (procPlg *procPlugin) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, error) {
-	metricTypes := []plugin.MetricType{}
+func (procPlg *procPlugin) GetMetricTypes(cfg plugin.Config) ([]plugin.Metric, error) {
+	metricTypes := []plugin.Metric{}
 	// build metric types from process metric names
 	for metricName, label := range metricNames {
-		metricType := plugin.MetricType{
-			Namespace_: core.NewNamespace(pluginVendor, fs, pluginName).
-				AddDynamicElement("process_name", "name of the running process").
-				AddStaticElements(metricName),
-			Config_:      cfg.ConfigDataNode,
-			Description_: label.description,
-			Unit_:        label.unit,
-		}
-		metricTypes = append(metricTypes, metricType)
-	}
-	// build metric types from process states
-	for _, state := range States.Values() {
-		metricType := plugin.MetricType{
-			Namespace_:   core.NewNamespace(pluginVendor, fs, pluginName, state),
-			Config_:      cfg.ConfigDataNode,
-			Description_: fmt.Sprintf("Number of processes in %s state", state),
-			Unit_:        "",
+		switch label.category {
+		case "pid":
+			metricTypes = append(metricTypes, plugin.Metric{
+				Namespace: plugin.NewNamespace(pluginVendor, fs, PluginName, "process").
+					AddDynamicElement("process_name", "name of the running process").
+					AddDynamicElement("process_pid", "indentifier of the running process").
```

Comment:
I am not sure if all these processes under this metric can be described as "running". Some of them might be in different state (sleeping, etc.). Could you confirm if that is true or not? If yes, consider removing "running" from element description.